File: app/src/main/java/com/app/cheffyuser/cart/TabsFragment.java
package com.app.cheffyuser.cart;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.fragment.app.Fragment;
import androidx.viewpager.widget.ViewPager;

import com.app.cheffyuser.R;
import com.app.cheffyuser.cart.adapter.TabsAdapter;
import com.google.android.material.tabs.TabLayout;


public class TabsFragment extends Fragment {

    public TabsFragment() {
        // Required empty public constructor
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {


        View view = inflater.inflate(R.layout.fragment_tabs, container, false);

        TabLayout tabLayout = (TabLayout)view. findViewById(R.id.tab_layout);
        tabLayout.addTab(tabLayout.newTab().setText("Add Cart"));
        tabLayout.addTab(tabLayout.newTab().setText("Custom Order"));
        tabLayout.addTab(tabLayout.newTab().setText("Tracking Order"));
        tabLayout.addTab(tabLayout.newTab().setText("Delivery Complete"));
        tabLayout.setTabGravity(TabLayout.GRAVITY_FILL);

        final ViewPager viewPager =(ViewPager)view.findViewById(R.id.view_pager);
        TabsAdapter tabsAdapter = new TabsAdapter(getActivity().getSupportFragmentManager(), tabLayout.getTabCount());
        viewPager.setAdapter(tabsAdapter);
        viewPager.addOnPageChangeListener(new TabLayout.TabLayoutOnPageChangeListener(tabLayout));
        tabLayout.setOnTabSelectedListener(new TabLayout.OnTabSelectedListener() {
            @Override
            public void onTabSelected(TabLayout.Tab tab) {
                viewPager.setCurrentItem(tab.getPosition());
            }
            @Override
            public void onTabUnselected(TabLayout.Tab tab) {

            }
            @Override
            public void onTabReselected(TabLayout.Tab tab) {

            }
        });



        return view;
    }




    @Override
    public void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
    }
}

File: app/src/main/java/com/app/cheffyuser/cart/TabsFragment.kt
package com.app.cheffyuser.cart

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.viewpager.widget.ViewPager
import com.app.cheffyuser.R
import com.app.cheffyuser.cart.fragments.AddCartFragment
import com.app.cheffyuser.cart.fragments.CustomOrderFragment
import com.app.cheffyuser.cart.fragments.DeliveryFragment
import com.app.cheffyuser.cart.fragments.TrackingOrderFragment
import com.app.cheffyuser.home.adapter.CartTabAdapter
import com.app.cheffyuser.home.fragments.BaseFragment
import com.google.android.material.tabs.TabLayout
import kotlinx.android.synthetic.main.fragment_tabs.*


class TabsFragment : BaseFragment() {

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.fragment_tabs, container, false)
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)

        val adapter = CartTabAdapter(childFragmentManager)
        //setup vp
        //pager.setPagingEnabled(true)

        adapter.addFragments(AddCartFragment())
        adapter.addFragments(CustomOrderFragment())
        adapter.addFragments(TrackingOrderFragment())
        adapter.addFragments(DeliveryFragment())
        pager.adapter = adapter

        pager.offscreenPageLimit = 4
        pager.currentItem = 0

        tab_layout.tabGravity = TabLayout.GRAVITY_FILL
        tab_layout.setupWithViewPager(pager)

        pager.addOnPageChangeListener(object : ViewPager.OnPageChangeListener {
            override fun onPageScrollStateChanged(state: Int) {}

            override fun onPageScrolled(
                position: Int,
                positionOffset: Float,
                positionOffsetPixels: Int
            ) {
            }

            override fun onPageSelected(position: Int) {
                when (position) {
                    0 -> {
                        //TODO: refresh the viewmodels
                    }

                    1 -> {

                    }

                    2 -> {

                    }
                }
            }

        })

    }
}
